core/trino-spi/src/main/java/io/trino/spi/type/LongDecimalType.java

```diff
@@ -95,7 +98,12 @@ public Object getObjectValue(ConnectorSession session, Block block, int position
             return null;
         }
         Slice slice = getSlice(block, position);
-        return new SqlDecimal(decodeUnscaledValue(slice), getPrecision(), getScale());
+        BigInteger unscaledValue = decodeUnscaledValue(slice);
+        // sanity check: disallow "negative zero" in case we get a corrupted result from somewhere
+        if (unscaledValue.signum() == 0 && isNegative(slice)) {
```

Comment:
Make it a one line `checkArgument` statement

Reply:
Done, thanks